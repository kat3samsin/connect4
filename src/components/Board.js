import React, { Component } from 'react';
import {connect} from 'react-redux';
import Row  from './Row';

export class Board extends Component {
  render() {
    return (<div>
        <table className='board'>
          <tbody>
            {this.props.board.map((row, idx) => <Row key={idx} row={row} />)}
          </tbody>            
        </table>
      </div>);
  }
}

const mapStateToProps = (state) => {
  return state;
}
export default connect(mapStateToProps)(Board);
