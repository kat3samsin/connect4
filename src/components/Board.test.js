import React from 'react';
import { shallow } from 'enzyme';
import { Board } from './Board';

describe('Board', () => {
  let wrapper;

   beforeEach(() => {
     wrapper = shallow(<Board board={[[0, 0, 0], [0, 0, 0], [0, 0, 0]]} />);
   });

  it('renders without crashing', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('renders a div and table', () => {
    expect(wrapper.find('div')).toHaveLength(1);
    expect(wrapper.find('table')).toHaveLength(1);
    expect(wrapper.find('tbody')).toHaveLength(1);
  });

  it('renders 3 rows', () => {
    expect(wrapper.find('Row')).toHaveLength(3);
  });
});