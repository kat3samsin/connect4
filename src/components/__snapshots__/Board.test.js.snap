// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Board renders without crashing 1`] = `ShallowWrapper {}`;
